src/MizarData/include/MizarData/ActiveFunctionTimePerFrameComparator.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_ACTIVE_FUNCTION_TIME_PER_FRAME_COMPARATOR_H_
+#define MIZAR_DATA_ACTIVE_FUNCTION_TIME_PER_FRAME_COMPARATOR_H_
+
+#include <utility>
+
+#include "ClientData/ScopeStats.h"
+#include "MizarData/BaselineOrComparison.h"
+#include "MizarData/SamplingWithFrameTrackComparisonReport.h"
+#include "Statistics/Gaussian.h"
+#include "Statistics/StatisticsUtils.h"
+
+namespace orbit_mizar_data {
+
+// TODO(b/236130122) it should be moved to a new library
+// Implements the statistical hypothesis testing procedure aimed to test the equality of total
+// CPU-time of the sampled functions given the sampled rates and the frame track stats.
+// Under the assumption of equality the distribution of the statistic is approximated with normal
+// distribution.
+template <typename Counts, typename FrameTrackStats>
+class ActiveFunctionTimePerFrameComparatorTmpl {
+ public:
+  explicit ActiveFunctionTimePerFrameComparatorTmpl(
+      const Baseline<Counts>& baseline_counts,
+      const Baseline<FrameTrackStats>& baseline_frame_stats,
+      const Comparison<Counts>& comparison_counts,
+      const Comparison<FrameTrackStats>& comparison_frame_stats)
+      : baseline_counts_(baseline_counts),
+        baseline_frame_stats_(baseline_frame_stats),
+        comparison_counts_(comparison_counts),
+        comparison_frame_stats_(comparison_frame_stats) {}
+
+  [[nodiscard]] ComparisonResult Compare(SFID sfid) const {
+    const auto baseline_active_time =
```

Comment:
From the style guide:
>The fundamental rule is: use type deduction only to make the code clearer or safer, and do not use it merely to avoid the inconvenience of writing an explicit type. When judging whether the code is clearer, keep in mind that your readers are not necessarily on your team, or familiar with your project, so types that you and your reviewer experience as unnecessary clutter will very often provide useful information to others. For example, you can assume that the return type of make_unique<Foo>() is obvious, but the return type of MyWidgetFactory() probably isn't.

So for me when reading the code, the return type of `ActiveFunctionType` is not clear at all. Please avoid the use of auto here.